### DISCUSSION TITLE
io.to(socketid).emit() not working

### DISCUSSION THREAD

**Discussion by hackthedev** [socketio/socket.io#4449] [Q&A]
Hi.

Im having the following line:
`io.to(toKey).emit("sendInfo", "<response>" + socket.id + "</response>");`

toKey is the socket id of another connection. Im using chrome for client 1 and edge for client 2.

My problem is that there is nothing happening at all. Its seems to not get executed, but also is not throwing any error. Im executing io.to in the index.js File (server).

Updated to the latest node.js version as well and same for npm.

This is the code for the event
```
socket.on('sendInfo', function(message) {
        alert("Working");
        console.log("Sending information back to user " + fromRegex(message)[0]);
        socket.emit("sendBackRequestInfo", fromRegex(message)[0], getCookie("username"), getCookie("tag"), getCookie("key"));
    });
```

its not being triggered. If i do io.emit it works. same as if i do io.to(socket.id).emit(..) (my id). 

Its Node.js 16.16.0 and Socket.io.js v4.5

**Comment by FunXTime** [comment]
Hi, this shall help you.
![image](https://user-images.githubusercontent.com/70162974/193105059-120f8d2a-85f2-4716-9650-f01cb0015721.png)
(From https://socket.io/docs/v4/server-api/#sockettoroom example)

This method you attempted will NOT work as it will send it to everyone in the room.

**Reply by danilo-vieira** [reply]
In their code example he is using `io.to` not `socket.to` what means it was to work. Here im with same problem :/

**Comment by ghost** [comment]
Same problem here I am using Socket.IO version 3.x and I want to emit to a particular socket ( private messaging kind )
i use io.to(socketId).emit('event', {}) but client does not receive this event 

even though client is connected to the socket server

I am using node JS version 14.19.0 in backend and in frontend we have iOS app which does not receive this event

**Reply by CristianmMst** [reply]
¿Could you solve it?

**Reply by hackthedev** [reply]
The following code is working for me now
```
io.to(socket.id).emit("updateMemberList");
```

Im using socket.io v4.6.1

**Reply by lakshmipriya-maticz** [reply]
this is not working for me

**Reply by lakshmipriya-maticz** [reply]
I have a socket id to emit sepcific user

my code ,
io.to(socketid).emit("updatemsg","ok")

its not receieved on socketid user
@hackthedev

**Comment by Nour-Banoun** [comment]
after each socket connection add the user to his socket channnel 
```
io.on('connection', (socket) =>{
  socket.join(socket.id)
.
.
.
}
```
then you can emit to the socket id
```
socket.to(rec_socket_id).emit('event',{})
```

**Reply by darrachequesne** [reply]
Calling `socket.join(socket.id)` is already done for you by the library so it's not necessary.

Source: https://github.com/socketio/socket.io/blob/b56da8a99f4814d553064af175edcf747d5561d7/lib/socket.ts#L597-L600

**Comment by Oleh1Kuper** [comment]
I had the same problem, the code below works for me

server.js
`io.to(receiverSocketId).to(socket.id).emit('event', data)`
